tests-legacy/Unit/Classes/ObjectModelTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace LegacyTests\Unit\Classes;
+
+use LegacyTests\TestCase\UnitTestCase;
+use Alias;
+
+class ObjectModelTest extends UnitTestCase
```

Comment:
I dont feel good having a unit test modifying `Alias::$definition` in `setUp()` and `tearDown()` 🤔.

If multiple unit tests are run at the same time without process isolation, this might break another unit test using `Alias` right ? 🤔 

Can we think about one way to use a inoffensive ObjectModel for this test ? A dummy ObjectModel.

The very best would be to create the dummy class inside `setUp()` and delete it in `tearDown()` but I dont know if we can do this.

Reply:
Agree, please add a DummyObjectModel linked to these tests

Reply:
> Agree, please add a DummyObjectModel linked to these tests

It wont work or be autoloaded except if I put it on `classes` folder at the root of PrestaShop: you won't agree on this.

I can mock one, but mock a class to test one of its public method... ? Not sure at all it will proves something.

> If multiple unit tests are run at the same time without process isolation, this might break another unit test using Alias right ? :thinking:

Not at all, `setUp` and `tearDown` will be executed only for assertions of this class: what is down in `setUp` is reverted in `tearDown`.

I can eventually extends an existing one instead of using it: what do you think ?

Reply:
> I can eventually extends an existing one instead of using it: what do you think ?
> 
> ```
> class DummyObjectModel extends ObjectModel
> {
>       // alter the definition
> }
> ```
But how will you autoload it if only `classes` folder is being parsed by ObjectModel ?

I agree this is not easy to fix 🤔.

Reply:
> But how will you autoload it if only classes folder is being parsed by ObjectModel ?

Not really, because my DummyObject would be located in a namespaced file.

Also another alternative: anonymous class ?

```php
private function dummy()
    {
        return new class() extends ObjectModel {
             public $definition = [...]; // what i need to do.
        };
    }
```

Reply:
Anonymous class would be perfect 👍 absolutely no possible side-effect I can think of that could affect other running tests